management-core/src/main/java/com/ns/task/repositories/ProductRepositoryCustom.java

```diff
@@ -2,10 +2,11 @@
 
 import com.ns.task.entities.ProductEntity;
 
+import java.sql.SQLIntegrityConstraintViolationException;
 import java.util.List;
 
 public interface ProductRepositoryCustom {
     List<ProductEntity> retrieveProducts();
 
-    ProductEntity saveProduct(ProductEntity product);
+    ProductEntity saveProduct(ProductEntity product) throws SQLIntegrityConstraintViolationException;
```

Comment:
Why this SQLIntegrityConstraintViolationException?

Reply:
changed to DataIntegrityViolationException, this one is the abstraction exception that spring throws, I am not sure if this is the correct one